docs/elements.md

```diff
@@ -1256,6 +1257,63 @@ See the [`pl-drawing` documentation](pl-drawing/index.md) for details.
 
 -----
 
+## `pl-overlay` element
+
+The overlay element allows existing PrairieLearn and HTML elements to be layered on top of one another in arbitrary positions.
+
+#### Sample Element
+
+![](elements/pl-overlay.png)
+
+```html
+<pl-overlay width="400" height="400" clip="false">
+    <pl-location left="0" top="0" halign="left" valign="top">
+        <pl-drawing width="398" height="398" hide-answer-panel="false">
+            <pl-drawing-initial>
+                <pl-triangle x1="50" y1="350" x2="350" y2="350" x3="350" y3="50"></pl-triangle>
+            </pl-drawing-initial>
+        </pl-drawing>
+    </pl-location>
+    <pl-location left="200" top="375">
+        $$3$$
+    </pl-location>
+    <pl-location left="375" top="200">
+        $$3$$
+    </pl-location>
+    <pl-location left="170" top="170">
+        <pl-number-input answers-name="c" show-help-text="false" show-placeholder="false" size="1"></pl-number-input>
+    </pl-location>
+</pl-overlay>
+```
+
+#### `pl-overlay` Customizations
+
+Attribute | Type | Default | Description
+--- | --- | --- | ---
+`width` | float | - | The width of the overlay canvas in pixels.
+`height` | float | - | The height of the overlay canvas in pixels.
+`clip` | boolean | True | If true, children will be cut off when exceeding overlay boundaries.
+
+#### `pl-location` Customizations
+
+Attribute | Type | Default | Description
+--- | --- | --- | ---
+`left` | float | - | The x coordinate of the child element (relative to the left of the overlay)
+`top` | float | - | The y coordinate of the child element (relative to the top of the overlay)
+`right` | float | - | The x coordinate of the child element (relative to the right of the overlay)
+`bottom` | float | - | The y coordinate of the child element (relative to the bottom of the overlay)
+`valign` | string | `middle` | Specifies the vertical alignment of the contents.  Can be one of `top`, `middle`, or `bottom`.
+`halign` | string | `center` | Specifies the horizontal alignment of the contents.  Can be one of `left`, `center`, or `right`.
```

Comment:
A possible future enhancement would be to have another child type of `<pl-base-image>` from which we automatically pull the width/height. At the moment it's a bit of messing around to get an image, find it's dimensions, set those explicitly as width/height for `pl-overlay`, include the image at position (0,0), etc.

Or maybe we should just have a `base-image="mapzoom.png"` attribute for `pl-overlay`? I'm not sure.

Another thought: it would be really cool to have a demo question which has an image with many locations on it. For example, the US map of states. The Python file could have a big list of (x,y) locations and state abbreviations, and we could show how to randomize the placement of the text input box over a state and the setting of the correct answer. This would be an example of a "label parts of an image" question.

Another way to set up a random problem like this would be to choose five random elements from the big list of (x,y,name) tuples, use the first one to overlay the simple text "State X" on the image, and use the five elements as MC alternatives (with the first one marked as correct).

I think both of these are good general patterns for labeling questions and we should probably include both.

Reply:
It occurs to me that we don't have a `pl-dropdown` element. If we did have one, then it would be cool to have a map of the states with five random states containing dropdowns with all 50 states listed in them. So the student has to pick the names of 5 different states, getting 20% of the points for each one that is correct.